### PR TITLE
Creating a group

### PR DESCRIPTION
**Pull request by MadhuriUttarkar**

### CODE REVIEW

File: src/main/java/org/madhuri/app/message/util/HibernateUtil.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	
+
+	/*public HibernateUtil() {
+		Configuration configuration = new Configuration();
+		configuration.configure("hibernate.cfg.xml");
+
+		sessionFactory = configuration.buildSessionFactory();
+	}*/
+
+	public static Session getSession() {
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/util/HibernateUtil.java]: this creates a session for every getSession call ?
How can you optimize to only create a session once at the starting of the app ?
something like singleton pattern ?

File: src/main/java/org/madhuri/app/message/handler/GroupHandler.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+@Path("/groups")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class GroupHandler {
+	
+	private GroupService groupService = new GroupService();
+	
+	@GET
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/handler/GroupHandler.java]: format the code

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    private Long groupid;
+
+    @Column(name="groupName")
+    private String groupName;
+
+    @Column(name="groupAdmin")
+    private String groupAdmin;
+
+    @Column(name="groupMessages")
+    private String groupMessages;
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: what is groupMessages ?

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name="groupid")
+    private Long groupid;
+
+    @Column(name="groupName")
+    private String groupName;
+
+    @Column(name="groupAdmin")
+    private String groupAdmin;
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: is Admin not a profileId ?

**Review by dwarakauttarkar — COMMENTED** [review]
take a look at comments

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java (hunk 4 above)
Comment (MadhuriUttarkar) [reply src/main/java/org/madhuri/app/message/model/MessageGroup.java]: groupAdmin is the owner of the group

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java (hunk 3 above)
Comment (MadhuriUttarkar) [reply src/main/java/org/madhuri/app/message/model/MessageGroup.java]: Messages in the group

**Review by dwarakauttarkar — APPROVED** [review]

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.Date;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "MessageGroup")
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: channel or group

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+public class MessageGroup {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "groupid")
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: id

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+	private Long groupid;
+
+	@Column(name = "groupName")
+	private String groupName;
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: name

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+	@Column(name = "groupAdmin")
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: admin_id

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+	private String groupAdmin;
+
+	@Column(name = "groupMessages")
+	private String groupMessages;
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: welcome_message

File: src/main/java/org/madhuri/app/message/model/MessageGroup.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+	@Column(name = "created")
+	private Date created;
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/MessageGroup.java]: created_at

**Review by dwarakauttarkar — CHANGES_REQUESTED** [review]

File: src/main/java/org/madhuri/app/message/model/Channel.java [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+	private Long admin_id;
+
+	@Column(name = "welcome_message")
+	private String welcome_message;
+
+	@Column(name = "created_at")
+	private Date created_at;
+	
+	@Column(name="created_by")
+	private String created_by;
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/Channel.java]: Is created_by a user ? Then why not use id here ?

File: src/main/java/org/madhuri/app/message/model/Channel.java [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+
+	public void setId(Long id) {
+		this.id = id;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public void setAdmin_id(Long admin_id) {
```
Comment (dwarakauttarkar) [inline src/main/java/org/madhuri/app/message/model/Channel.java]: Method names are never snake case in Java...

**Review by dwarakauttarkar — APPROVED** [review]

**Pull request merged**